test/tests/dom/waitForDocumentBody.js

```diff
@@ -0,0 +1,42 @@
+/* @flow */
+
+import { waitForDocumentBody } from '../../../src/dom';
+import { memoize } from '../../../src/util';
+
+describe('waitForDocumentReady cases', () => {
+    const oldBody = document.body;
+    const testBody = document.createElement('body');
+
+    beforeEach(memoize.clear);
+
+    afterEach(() => {
+        document.body = oldBody;
+    });
+    it('should resolve when body is present', async () => {
+        try {
+            document.readyState = 'complete';
+            document.body = testBody;
+            await waitForDocumentBody();
+        } catch (err) {
```

Comment:
I think we can remove this try/catch -- if there's an error thrown, it's probably best for the test to fail with the original error.